client/services/factory.js

```diff
@@ -2,25 +2,23 @@ var app = angular.module('eir.factory', []);
 
 
 app.service('fileUpload', function($http) {
-  this.uploadFileToUrl = function(fields, uploadUrl) {
-    uploadUrl = uploadUrl || 'classes/patients';
-    var fd = new FormData();
-    
-    // append each field to the FormData object
-    for(var field in fields) {
-      fd.append(field, fields[field]);
+  return {
+    uploadFileToUrl:function(fields, uploadUrl) {
+      uploadUrl = uploadUrl || 'classes/patients';
+      var fd = new FormData();
+      
+      // append each field to the FormData object
+      for(var field in fields) {
+        fd.append(field, fields[field]);
+      }
+      
+      return $http.post(uploadUrl, fd, {
```

Comment:
@benjaminhoffman here I am just returning the http factory... this returns a promise that is thenable and catchable